packages/@expo/fingerprint/src/Config.ts

```diff
@@ -0,0 +1,86 @@
+import fs from 'fs/promises';
+import path from 'path';
+
+import type { Config } from './Fingerprint.types';
+
+const CONFIG_FILES = ['fingerprint.config.js', 'fingerprint.config.cjs'];
+
+const debug = require('debug')('expo:fingerprint:Config');
+
+/**
+ * Load the fingerprint.config.js from project root.
+ * @param projectRoot The project root directory.
+ * @param silent Whether to mute console logs when loading the config. This is useful for expo-updates integration and makes sure the JSON output is valid.
+ * @returns The loaded config or null if no config file was found.
+ */
+export async function loadConfigAsync(
+  projectRoot: string,
+  silent: boolean = false
+): Promise<Config | null> {
+  const configFile = await resolveConfigFileAsync(projectRoot);
+  if (!configFile) {
+    return null;
+  }
+  debug('Resolved config file:', configFile);
+
+  const unregisterMuteLogs = silent ? muteLogs() : null;
+  let rawConfig;
+  try {
+    rawConfig = require(configFile);
+  } catch (e: unknown) {
+    debug('Error loading config file:', e);
+    rawConfig = {};
+  }
+  unregisterMuteLogs?.();
+
+  const supportedConfigKeys: (keyof Config)[] = [
+    'concurrentIoLimit',
+    'hashAlgorithm',
+    'extraSources',
```

Comment:
Is the value for this key serializable to JSON? (can it be specified)

Reply:
mostly, the config is just configurable Options. probably [Buffer](https://github.com/expo/expo/blob/55e4ad4da9aba74562be03baa5d14bb03b532f74/packages/%40expo/fingerprint/src/Fingerprint.types.ts#L111) is not serializable but can use string instead.

would make the implementation simpler first without validator or parser. let's add it later if necessary.